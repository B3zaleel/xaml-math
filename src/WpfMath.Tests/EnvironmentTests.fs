module WpfMath.Tests.EnvironmentTests

open Xunit

open WpfMath.Exceptions
open WpfMath.Tests.ApprovalTestUtils
open WpfMath.Tests.Utils

[<Fact>]
let pMatrixEnvironment(): unit =
    verifyParseResult @"\begin{pmatrix}{line 1}\\line 2\end{pmatrix}"

[<Fact>]
let nestedEnvironment(): unit =
    verifyParseResult @"\begin{pmatrix}line 1\\\begin{pmatrix}line x\end{pmatrix}\end{pmatrix}"

[<Fact>]
let nestedMatrix(): unit =
    verifyParseResult @"\begin{pmatrix}line 1\\\pmatrix{line x & line y}\end{pmatrix}"

[<Fact>]
let ``Empty environment name should trigger an exception``(): unit =
    let markup = @"\begin{}"
    let ex = assertParseThrows<TexParseException> markup
    Assert.Equal(@"Empty environment name for the \begin command.", ex.Message)
