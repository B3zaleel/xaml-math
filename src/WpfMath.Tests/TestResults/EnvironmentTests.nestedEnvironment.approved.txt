{
  "RootAtom": {
    "[AtomType]": "FencedAtom",
    "BaseAtom": {
      "[AtomType]": "MatrixAtom",
      "HorizontalPadding": 0.35,
      "MatrixCellAlignment": "Center",
      "MatrixCells": [
        [
          {
            "[AtomType]": "RowAtom",
            "Elements": [
              {
                "[AtomType]": "CharAtom",
                "Character": "l",
                "IsTextSymbol": false,
                "Source": {
                  "End": 16,
                  "Length": 1,
                  "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                  "SourceName": "User input",
                  "Start": 15
                },
                "TextStyle": null,
                "Type": "Ordinary"
              },
              {
                "[AtomType]": "CharAtom",
                "Character": "i",
                "IsTextSymbol": false,
                "Source": {
                  "End": 17,
                  "Length": 1,
                  "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                  "SourceName": "User input",
                  "Start": 16
                },
                "TextStyle": null,
                "Type": "Ordinary"
              },
              {
                "[AtomType]": "CharAtom",
                "Character": "n",
                "IsTextSymbol": false,
                "Source": {
                  "End": 18,
                  "Length": 1,
                  "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                  "SourceName": "User input",
                  "Start": 17
                },
                "TextStyle": null,
                "Type": "Ordinary"
              },
              {
                "[AtomType]": "CharAtom",
                "Character": "e",
                "IsTextSymbol": false,
                "Source": {
                  "End": 19,
                  "Length": 1,
                  "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                  "SourceName": "User input",
                  "Start": 18
                },
                "TextStyle": null,
                "Type": "Ordinary"
              },
              {
                "[AtomType]": "CharAtom",
                "Character": "1",
                "IsTextSymbol": false,
                "Source": {
                  "End": 21,
                  "Length": 1,
                  "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                  "SourceName": "User input",
                  "Start": 20
                },
                "TextStyle": null,
                "Type": "Ordinary"
              }
            ],
            "PreviousAtom": null,
            "Source": {
              "End": 21,
              "Length": 6,
              "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
              "SourceName": "User input",
              "Start": 15
            },
            "Type": "Ordinary"
          }
        ],
        [
          {
            "[AtomType]": "FencedAtom",
            "BaseAtom": {
              "[AtomType]": "MatrixAtom",
              "HorizontalPadding": 0.35,
              "MatrixCellAlignment": "Center",
              "MatrixCells": [
                [
                  {
                    "[AtomType]": "RowAtom",
                    "Elements": [
                      {
                        "[AtomType]": "CharAtom",
                        "Character": "l",
                        "IsTextSymbol": false,
                        "Source": {
                          "End": 39,
                          "Length": 1,
                          "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                          "SourceName": "User input",
                          "Start": 38
                        },
                        "TextStyle": null,
                        "Type": "Ordinary"
                      },
                      {
                        "[AtomType]": "CharAtom",
                        "Character": "i",
                        "IsTextSymbol": false,
                        "Source": {
                          "End": 40,
                          "Length": 1,
                          "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                          "SourceName": "User input",
                          "Start": 39
                        },
                        "TextStyle": null,
                        "Type": "Ordinary"
                      },
                      {
                        "[AtomType]": "CharAtom",
                        "Character": "n",
                        "IsTextSymbol": false,
                        "Source": {
                          "End": 41,
                          "Length": 1,
                          "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                          "SourceName": "User input",
                          "Start": 40
                        },
                        "TextStyle": null,
                        "Type": "Ordinary"
                      },
                      {
                        "[AtomType]": "CharAtom",
                        "Character": "e",
                        "IsTextSymbol": false,
                        "Source": {
                          "End": 42,
                          "Length": 1,
                          "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                          "SourceName": "User input",
                          "Start": 41
                        },
                        "TextStyle": null,
                        "Type": "Ordinary"
                      },
                      {
                        "[AtomType]": "CharAtom",
                        "Character": "x",
                        "IsTextSymbol": false,
                        "Source": {
                          "End": 44,
                          "Length": 1,
                          "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                          "SourceName": "User input",
                          "Start": 43
                        },
                        "TextStyle": null,
                        "Type": "Ordinary"
                      }
                    ],
                    "PreviousAtom": null,
                    "Source": {
                      "End": 44,
                      "Length": 6,
                      "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                      "SourceName": "User input",
                      "Start": 38
                    },
                    "Type": "Ordinary"
                  }
                ]
              ],
              "Source": {
                "End": 57,
                "Length": 33,
                "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
                "SourceName": "User input",
                "Start": 24
              },
              "Type": "Ordinary",
              "VerticalPadding": 0.35
            },
            "LeftDelimeter": {
              "[AtomType]": "SymbolAtom",
              "IsDelimeter": true,
              "IsTextSymbol": false,
              "Name": "lbrack",
              "Source": null,
              "Type": "Opening"
            },
            "RightDelimeter": {
              "[AtomType]": "SymbolAtom",
              "IsDelimeter": true,
              "IsTextSymbol": false,
              "Name": "rbrack",
              "Source": null,
              "Type": "Closing"
            },
            "Source": {
              "End": 57,
              "Length": 33,
              "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
              "SourceName": "User input",
              "Start": 24
            },
            "Type": "Ordinary"
          }
        ]
      ],
      "Source": {
        "End": 70,
        "Length": 69,
        "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
        "SourceName": "User input",
        "Start": 1
      },
      "Type": "Ordinary",
      "VerticalPadding": 0.35
    },
    "LeftDelimeter": {
      "[AtomType]": "SymbolAtom",
      "IsDelimeter": true,
      "IsTextSymbol": false,
      "Name": "lbrack",
      "Source": null,
      "Type": "Opening"
    },
    "RightDelimeter": {
      "[AtomType]": "SymbolAtom",
      "IsDelimeter": true,
      "IsTextSymbol": false,
      "Name": "rbrack",
      "Source": null,
      "Type": "Closing"
    },
    "Source": {
      "End": 70,
      "Length": 69,
      "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
      "SourceName": "User input",
      "Start": 1
    },
    "Type": "Ordinary"
  },
  "Source": {
    "End": 70,
    "Length": 70,
    "Source": "\\begin{pmatrix}line 1\\\\\\begin{pmatrix}line x\\end{pmatrix}\\end{pmatrix}",
    "SourceName": "User input",
    "Start": 0
  },
  "TextStyle": null
}